backend/tests/controllers/artefacts/test_artefacts_search.py

```diff
@@ -0,0 +1,220 @@
+    assert response.status_code == 200
+    data = response.json()
+    assert "artefacts" in data
+    assert isinstance(data["artefacts"], list)
+
+
+def test_search_artefacts_no_query(test_client: TestClient, generator: DataGenerator):
+    """Test searching without query returns all artefacts"""
+    # Create some artefacts
+    unique_marker = uuid.uuid4().hex[:8]
+    artefact1 = generator.gen_artefact(name=f"test_artefact_1_{unique_marker}")
+    artefact2 = generator.gen_artefact(name=f"test_artefact_2_{unique_marker}")
+
+    response = test_client.get("/v1/artefacts/search")
+
+    assert response.status_code == 200
+    data = response.json()
+    artefacts = data["artefacts"]
+
+    # Should contain our created artefacts
+    assert artefact1.name in artefacts
+    assert artefact2.name in artefacts
+
+
+def test_search_artefacts_with_query(test_client: TestClient, generator: DataGenerator):
+    """Test searching with query filters results"""
+    unique_marker = uuid.uuid4().hex[:8]
+
+    # Create artefacts with specific patterns
+    target_artefact = generator.gen_artefact(name=f"special_search_{unique_marker}")
+    other_artefact = generator.gen_artefact(name=f"other_artefact_{unique_marker}")
+
+    # Search for the specific pattern
+    response = test_client.get("/v1/artefacts/search?q=special_search")
+
+    assert response.status_code == 200
+    data = response.json()
+    artefacts = data["artefacts"]
+
+    # Should only find the target artefact
+    assert target_artefact.name in artefacts
+    assert other_artefact.name not in artefacts
+
+
+def test_search_artefacts_case_insensitive(
+    test_client: TestClient, generator: DataGenerator
+):
+    """Test that search is case-insensitive"""
+    unique_marker = uuid.uuid4().hex[:8]
+    artefact_name = f"MixedCase_Artefact_{unique_marker}"
+
+    generator.gen_artefact(name=artefact_name)
+
+    # Search with different cases
+    for search_term in ["mixedcase", "MIXEDCASE", "MixedCase"]:
+        response = test_client.get(f"/v1/artefacts/search?q={search_term}")
+        assert response.status_code == 200
+        artefacts = response.json()["artefacts"]
+        assert artefact_name in artefacts
+
+
+def test_search_artefacts_partial_match(
+    test_client: TestClient, generator: DataGenerator
+):
+    """Test that search supports partial matching"""
+    unique_marker = uuid.uuid4().hex[:8]
+    artefact_name = f"ubuntu_desktop_{unique_marker}"
+
+    generator.gen_artefact(name=artefact_name)
+
+    # Search with partial term
+    response = test_client.get("/v1/artefacts/search?q=desktop")
+
+    assert response.status_code == 200
+    artefacts = response.json()["artefacts"]
+    assert artefact_name in artefacts
+
+
+def test_search_artefacts_pagination(test_client: TestClient, generator: DataGenerator):
+    """Test that pagination works correctly"""
+    unique_marker = uuid.uuid4().hex[:8]
+    names = []
+
+    # Create 10 artefacts
```

Comment:
I feel like the whole PR contains too many comments that are redundant (like this one). could you please make sure you keep only the important ones?